pubsub/pubsub.go

```diff
@@ -40,3 +54,26 @@ func Ping(pubsub PubSub) error {
 		return fmt.Errorf("ping is not implemented by this pubsub")
 	}
 }
+
+// DefaultBulkMessager is default implemnetation for BukMessager
+type DefaultBulkMessager struct {
+	p PubSub
+}
+
+// NewDefaultBulkMessager to create new DefaultBulkMessager for a PubSub
+func NewDefaultBulkMessager(pubsub PubSub) DefaultBulkMessager {
```

Comment:
Why do we need this and then embed this in every component? Couldn't the runtime do a runtime check if the component implements the `BulkMessager` interface?

Reply:
One of the idea was to have a default bulk message implementation which provides a small perf improvement based on one single request from app to runtime even though it might be split into multiple requests between runtime and broker. These are for components that do not implement bulk publish interface. 

Similarly you have a case made for subscribe as well. That is why this interface and default implementation is done here. 
@ItalyPaleAle

Reply:
This has been defined here similar to the bulk state apis..

Reply:
I will leave it to a maintainer to decide. Personally, I think this would be best in the runtime, but I see the precedent.

Reply:
It is taken care of. No more mandatory for each pubsub to implement bulk pubsub APIs and runtime will check if component implements, else call default implementation - default is part of respective bulk publish and bulk subscribe PRs. Also, two different interfaces to let BulkPublish and BulkSubscribe remain independent.